cranelift/codegen/src/machinst/lower.rs

```diff
@@ -57,141 +57,11 @@ impl InstColor {
     }
 }
 
-/// A context that machine-specific lowering code can use to emit lowered
-/// instructions. This is the view of the machine-independent per-function
-/// lowering context that is seen by the machine backend.
-pub trait LowerCtx {
-    /// The instruction type for which this lowering framework is instantiated.
-    type I: VCodeInst;
-
-    fn dfg(&self) -> &DataFlowGraph;
-
-    // Function-level queries:
```

Comment:
FWIW, I found these "section headers" kind of useful when I spent more time looking at the lowering API. Could we keep them around by making them doc-comments on separate `impl` blocks? Something like:

```rust
/// Function-level query API on Lower.
impl Lower { ... }

/// Instruction-level query API on Lower.
impl Lower { ... }

/// Instruction input/output query API on Lower.
impl Lower { ... }

/// Codegen action API on Lower.
impl Lower { ... }
```